src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,328 @@
+
+       for pe in protectedEntities:
+          fad_records_iap = []
+          if pe['type'] == 'FSS':
+           
+             # Find the FSS records with OOBE protection
+             fss_with_oobe = self.findFSSInIAPwithOOBE(protectedEntities)
+             if fss_with_oobe :
+                # Call FSS purge list reference model to fetch cbsd to be purged. output:List of CBSDs
+                cbsd_purge_list = FSSPurgeListRefModel(fad_records,fssOobe)
+ 
+                # Remove purged cbsd from fad record
+                fad_records_iap = removePurgeCbsd(fad_records,cbsd_purge_list)
+             else:
+                fad_records_iap = fad_records
+          #Call IAP reference model, fad_records contains fad_UUT and fad_SAS_TH
+          #To-Do : If there is any change in the API defination, parameters need to be updated
+          iap_refmodel_margin_list = IapReferenceModel(config['headroom'], pe,fad_records_iap)
+ 
+
+       # Step 10 : Register N(2,k)CBSDs with SAS UUT
+       # Use DP objects to get CBSD registered
+       # creation of Domain Proxy Object
+       dp_cert = os.path.join('certs', 'dp_client.cert')
+       dp_key = os.path.join('certs', 'dp_client.key')
+       reg_records = config['cbsdRecords'][k]['registrationRequests']
+       grant_records = config['cbsdRecords'][k]['grantRequests']	
+       #dp.initialize triggers the registration and grant for the records passes as parameters
+       dp = DomainProxy(dp_cert,dp_key,reg_records,grant_records)
+
+     
+       # Step 11 : Send heartbeat request	
+       dp.heartbeat_request_for_all_grants()
+       
+       # Step 12 : Invoke Aggregate Interference Model
+	
+       cbsd_agg_list = dp.get_all_cbsd_objects_with_atleast_one_grant()
+
+       for pe in protectedEntities:
+ 
+          #invoking Aggregate Interference API,
+          agg_interference_list = AggregateInterferenceModel(cbsd_agg_list,pe)
+      
+          #invoking Aggregate Interference API,with IAP reference model EIRP values
+          # This is dummy API, need clarity on the parameters to be passed  
+          agg_interference_iap_list = AggregateInterferenceModelWithIAP(cbsd_agg_list,pe)
+          
+          #check MCP.1
+          checkMCP_IAP(pe,agg_interference_list,agg_interference_iap_list,pre_iap_headroom,leftOverAlloc)
+
+       # Step 13 : Configure SAS Harness with FAD,trigger FAD generation
+       self.configureFadForEachSASTH(k+1)
+       for N in len(config['sas_hareness']):
+          fad_object_th[N].initiate_full_activity_dump()
+
+
+       
+       #Step 14: Trigger Full Activty Dump to UUT 
+       self._sas_admin.TriggerFullActivityDump() # triggers to UUT
+
+       # Step 15 : Pull FAD records from SAS UUT
+       fad_uut_object.initiate_full_activity_dump()
+       
+       # Step 16 : Trigger CPAS and wait for its completion
+       self.self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+       
+       # Step 17 : Call IAP reference model
```

Comment:
Since this occurs in steps 9 and 17, it's probably best to abstract it into a function to ensure consistency.

Reply:
Updated,   Invoked pre-IAP module